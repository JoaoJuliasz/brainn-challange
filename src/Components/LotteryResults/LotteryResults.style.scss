.lottery-results-container {
    height: 100%;
    display: flex;
    align-items: center;
    position: relative;

    .lottery-results-numbers {
        display: flex;
        align-items: center;
        flex-wrap: wrap;
        height: 45%;
        width: 100%;
        z-index: 2;

        .justify-content {
            justify-content: space-around;
        }

        .numbers-container {
            flex: 0 0 16.66%;

            .number {
                background: #fff;
                font-size: 27px;
                border-radius: 50%;
                box-shadow: 5px 3px 33px -12px #000000;
                min-width: 82px;
                max-height: 82px;
                min-height: 82px;
                width: 75px;
                display: flex;
                align-items: center;
                justify-content: center;

                @media (max-width: 510px) {
                    min-width: 60px;
                    max-height: 60px;
                    min-height: 60px;
                    width: 60px;
                    justify-content: center;
                    margin: 5px;
                }
            }
        }

        @media (max-width: 860px) {
            height: 100% !important;
            justify-content: center !important;
            position: absolute;
            top: -20px;
        }
    }
}
