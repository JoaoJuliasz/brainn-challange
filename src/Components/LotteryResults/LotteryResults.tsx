import React from 'react';
import { Contest } from '../dataTypes/types';
// @ts-ignore
import './LotteryResults.style.scss'

type LotteryResultsType = {
    contest: Contest
}

const LotteryResults = ({ contest }: LotteryResultsType) => {
    return (
        <div className="lottery-results-container">
            <div className={`lottery-results-numbers ${contest.numeros?.length <= 6 && 'justify-content'}`}>
                {contest && contest.numeros?.map(number =>
                    <div key={number} className="numbers-container">
                        <div className="number">{number}</div>
                    </div>)}
            </div>
        </div>
    );
};

export default LotteryResults;