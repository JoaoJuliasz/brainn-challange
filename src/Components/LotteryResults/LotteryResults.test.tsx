import React from 'react'
// @ts-ignore
import { shallow } from 'enzyme'
import LotteryResults from './LotteryResults'

describe('LotteryResults',() => {
    it('expect to render LotteryResults component', () => {
        const mockContest = {
            id: '6',
            loteria: 456,
            numeros: ['1', '4', '5'],
            data: '2021-04-20T00:28:09.426Z'
        }
        const wrapper = shallow(<LotteryResults contest={mockContest} />)
        expect(wrapper).toMatchSnapshot()
    })
})