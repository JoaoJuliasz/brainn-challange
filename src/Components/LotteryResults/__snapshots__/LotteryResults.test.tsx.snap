// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LotteryResults expect to render LotteryResults component 1`] = `
<div
  className="lottery-results-container"
>
  <div
    className="lottery-results-numbers justify-content"
  >
    <div
      className="numbers-container"
      key="1"
    >
      <div
        className="number"
      >
        1
      </div>
    </div>
    <div
      className="numbers-container"
      key="4"
    >
      <div
        className="number"
      >
        4
      </div>
    </div>
    <div
      className="numbers-container"
      key="5"
    >
      <div
        className="number"
      >
        5
      </div>
    </div>
  </div>
</div>
`;
